src/formatDate.js

```diff
@@ -50,7 +50,60 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dataArray = date.split(fromFormat[3]);
+  const result = [...dataArray];
+
+  // replace day
+  const dayIndexFrom = fromFormat.indexOf('DD');
+  const dayIndexTo = toFormat.indexOf('DD');
+
+  if (dayIndexFrom !== dayIndexTo) {
+    result[dayIndexTo] = dataArray[dayIndexFrom];
+  }
+
+  // replace month
+  const monthIndexFrom = fromFormat.indexOf('MM');
+  const monthIndexTo = toFormat.indexOf('MM');
+
+  if (monthIndexFrom !== monthIndexTo) {
+    result[monthIndexTo] = dataArray[monthIndexFrom];
+  }
+
+  // replace year
+  let yearIndexFrom = fromFormat.indexOf('YY');
+  let yearIndexTo = toFormat.indexOf('YY');
+
+  if (yearIndexFrom === -1) {
+    yearIndexFrom = fromFormat.indexOf('YYYY');
+  }
+
+  if (yearIndexTo === -1) {
+    yearIndexTo = toFormat.indexOf('YYYY');
+  }
+
+  if (yearIndexFrom !== yearIndexTo) {
+    result[yearIndexTo] = dataArray[yearIndexFrom];
+  }
+
+  // replace 2-char year to 4-char
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const indexYearFrom = fromFormat.indexOf('YY');
+
+    if (result[indexYearFrom] >= 30) {
+      result[indexYearFrom] = 19 + result[indexYearFrom];
+    } else {
+      result[indexYearFrom] = 20 + result[indexYearFrom];
+    }
+  }
+
+  // replace 4-char year to 2-char
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    const indexYearFrom = fromFormat.indexOf('YYYY');
+
+    result[indexYearFrom] = result[indexYearFrom].substring(2, 4);
+  }
+
+  return result.join(toFormat[3]);
```

Comment:
Lets make it shorter.
Refactor using loop, switch, ternary operator.
If you have questions ask it in fe_chart